syntax = "proto3";

package gateway;

//import "google/protobuf/empty.proto";

message MenuResponse {
    repeated Sandwiche sandwiches = 1;
    repeated DishMade dishMades = 2;
    repeated Drink drinks = 3;
    repeated Dessert desserts = 4;
}

message MenuRequest {
    int32 tableNumber = 1;
}

message Sandwiche {
    string id = 1;
    string name = 2;
    double price = 3;
    repeated string ingredients = 4;
}

message DishMade {
    string id = 1;
    string name = 2;
    double price = 3;
    repeated string ingredients = 4;
}

message Drink {
    string id = 1;
    string name = 2;
    double price = 3;
}

message Dessert {
    string id = 1;
    string name = 2;
    double price = 3;
}

message OrderRequest {
    repeated string id = 1;
    int32 tableNumber = 2;
}

message OrderResponse {
    double total = 1;
    int64 preparationTime = 2;
    repeated string foods = 3;
}

message CloseAccountRequest {
    int32 tableNumber = 1;
}

message CloseAccountResponse {
    int32 total = 1;
}

service Server {
    rpc GetMenu (MenuRequest) returns (MenuResponse) {}

    rpc CreateOrder (OrderRequest) returns (OrderResponse) {}

    rpc CloseAccount (CloseAccountRequest) returns (CloseAccountResponse) {}

    //rpc StatusOrder () returns () {}
}